### PR TITLE
Fix eps value for test_batchnorm_fallback to get rid of flakiness

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
As title. Ran 5000 times on CPU and passed, in the middle of 5000 verifications on GPU.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Shrink the eps value used

## Comments ##
Running 5000 verifications due to the large time cost (on average ~2s on CPU and ~8s on GPU)

### CODE REVIEW

File: tests/python/unittest/test_sparse_operator.py [hunk 1]
```diff
@@ -2121,16 +2121,16 @@ def test_batchnorm_fallback():
         mean_std = [mx.nd.array(rolling_mean).tostype(stype), mx.nd.array(rolling_std).tostype(stype)]
 
         test = mx.symbol.BatchNorm(data, fix_gamma=True)
-        assertRaises(MXNetError, check_numeric_gradient, test, in_location, mean_std, numeric_eps=1e-2, rtol=0.16, atol=1e-2)
+        assertRaises(MXNetError, check_numeric_gradient, test, in_location, mean_std, numeric_eps=1e-3, rtol=0.16, atol=1e-2)
```
Comment (anirudh2290) [inline tests/python/unittest/test_sparse_operator.py]: For batchnorm training test it is still 1e-2. Is that test expected to be flaky too ?

**Comment by haojin2** [comment]
@anirudh2290 Just verified that this can pass 5000 times on both CPU & GPU, should be good for merge.

File: tests/python/unittest/test_sparse_operator.py (hunk 1 above)
Comment (anirudh2290) [reply tests/python/unittest/test_sparse_operator.py]: just curious why we need to increase numeric_eps for this fallback test when it works fine for the training test even on the same seed.

Comment (haojin2) [reply tests/python/unittest/test_sparse_operator.py]: The original training test passed more than 10000 trials on both CPU & GPU

Comment (haojin2) [reply tests/python/unittest/test_sparse_operator.py]: Maybe we can merge this to stablize the CI first, and I'll do a deeper dive of this later. How do you like that?

Comment (anirudh2290) [reply tests/python/unittest/test_sparse_operator.py]: merging. please track in #11758
